mqtt/mqtt.js

```diff
@@ -169,20 +174,47 @@ aedes.authorizePublish = function (client, packet, publish) {
             return;
         }
     }
-    var channelTopic = elements.length ? baseTopic + '.' + elements.join('.') : baseTopic,
+
+    var last = elements[elements.length - 1],
+        format = 'text',
+        contentType = 'application/senml+json',
+        st = elements;
+    if (elements.length > 1) {
+        if (last === 'binary' || last === 'text') {
+            format = last;
+            contentType = elements[elements.length - 2].replace('_', '/').replace('-', '+');
```

Comment:
Please put some comments in this whole block. I can see what you are doing, but it might not be so easy for the newcomers.